principles.rst

```diff
@@ -0,0 +1,82 @@
+Principles for GHC
+==================
+
+This document lays out high-level principles for the evolution of GHC
+and the language it compiles. In much the way that we like to write
+*properties* of our functions -- instead of just writing down individual
+test cases -- this document strives to write *principles* of our proposals.
```

Comment:
Is it really a property of the proposal? Or rather a property of the design points that proposals move to? (Which would make them an invariant of the proposal, to stretch the analogy. Assuming a principled starting point).

Reply:
You may be onto something here. A principle here is more like an invariant than a property, I suppose. Care to suggest new wording?

Reply:
```suggestion
test cases -- this document strives to write *principles* that our proposals
ought to maintain (or build towards).
```